features/include/pcl/features/impl/good.hpp

```diff
@@ -0,0 +1,454 @@
+  
+  if (sum_all_bins != 0)
+  { 
+    size_t idx = 0; 
+    for(std::vector<unsigned int>::const_iterator it = histogram.begin(); it != histogram.end(); ++it)
+    {
+      normalized_histogram.at (idx) = (*it / sum_all_bins);
+      idx++;
+    }  
+  }
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////
+template<typename PointInT, int BinN> float
+pcl::GOODEstimation<PointInT, BinN>::viewpointEntropy (const std::vector<float> &normalized_histogram)
+{
+  ///NOTE: http://stats.stackexchange.com/questions/66108/why-is-entropy-maximised-when-the-probability-distribution-is-uniform
+  float entropy =0;  
+  for(std::vector<float>::const_iterator it = normalized_histogram.begin(); it != normalized_histogram.end(); ++it)
+  {
+    if (*it != 0)
+    {
+      float entropy_tmp = *it * log2 (*it);
+      entropy += entropy_tmp;
+    }
+  }  
+  return (-entropy); 
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////
+template<typename PointInT, int BinN> int 
+pcl::GOODEstimation<PointInT, BinN>::findMaxViewPointEntropy (const std::vector<float> &view_point_entropy)
+{
+  int index = 0;
+  std::vector<float>::const_iterator it;
+  it = std::max_element (view_point_entropy.begin(), view_point_entropy.end());
+  index = it - view_point_entropy.begin ();
+  return (index);
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////
+template<typename PointInT, int BinN> float
+pcl::GOODEstimation<PointInT, BinN>::varianceOfHistogram (const std::vector<float> &histogram)
+{
+  ///NOTE: https://people.richland.edu/james/lecture/m170/ch06-prb.html
+  ///NOTE: http://www.stat.yale.edu/Courses/1997-98/101/rvmnvar.htm  
+  
+  float mean = 0;
+  float variance = 0;
+  for (size_t i = 0; i < histogram.size(); i++)
+  {
+    mean += (i+1) * histogram.at(i);
+  }
+  
+  for (size_t i = 0; i < histogram.size(); i++)
+  {
+    variance += std::pow ((i+1)-mean , 2) * histogram.at(i);
+  }  
+  return (variance);  
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////
+template<typename PointInT, int BinN> void
+pcl::GOODEstimation<PointInT, BinN>::objectViewHistogram (int maximum_entropy_index, 
+                                                          const std::vector<std::vector<float> > &normalized_projected_views,
+                                                          std::vector<float> &sorted_normalized_projected_views, 
+                                                          std::string &name_of_sorted_projected_plane)
+{
+  float variance1 = 0;
+  float variance2 = 0;
+
+  sorted_normalized_projected_views.insert (sorted_normalized_projected_views.end(), normalized_projected_views.at(maximum_entropy_index).begin(), 
+                                            normalized_projected_views.at(maximum_entropy_index).end());    
```

Comment:
There's some code smell here.
- Do you know the size of `sorted_normalized_projected_views` in advance? If the answer is yes, you need to preallocate it in advance and use std::copy.
- Is this method's purpose to initialize `sorted_normalized_projected_views`? If so, the object should be constructed inside this function and passed out as a return value.